src/server/middleware/staticMiddlewareSubdirectoryFilter.js

```diff
@@ -27,6 +28,15 @@ fluid.defaults("sjrk.storyTelling.server.staticMiddlewareSubdirectoryFilter", {
     allowedSubdirectories: []
 });
 
+/**
+ * Restricts a given request based on whether the requested resource is contained
+ * within a directory that is on a list of allowed subdirectories
+ *
+ * @param {Objec} request - the kettle request
+ * @param {Object} allowedSubdirectories - the list of subdirectories from which to allow serving
+ *
+ * @return {Promise} - a fluid-flavoured promise that returns empty on resolve
```

Comment:
This is a generic comment for functional docs: please describe the error handling when applicable.

Reply:
I found [this comment](https://github.com/jsdoc/jsdoc/issues/1197#issuecomment-312948746) by the main developer of JSDoc, which suggests that a promise be documented only in terms of what is returned on resolution, not rejection. It would seem at some point in the future, the `@reject` syntax will be available, though this comment is nearly 3 years old.

I could use the `@error` tag, but this may be misleading since the function (in this case anyway) doesn't actually throw any exceptions. Perhaps I could add more detail in the `@return` portion of the documentation, despite the author's suggestion not to. Would that be satisfactory in the short term?

Reply:
Adding more detail in the @return portion of the documentation is a good idea.